process.c

```diff
@@ -3,60 +3,80 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <assert.h>
+#include <errno.h>
 
 #include "utils.h"
 #include "process.h"
 
 #define BUF_SIZE 100
 
-Process_t *create_process(char* pid) {
+int create_process(char *pid, Process_t **process) {
+  /*
+  pid: pid of the process whose information is to be read
+  process: pointer to the process to be created
+
+  returns: 0 upon successfull execution and 1 otherwise
+  */
+
   const char status_path_template[] = "/proc/%s/status";
-  const int status_path_template_len = strlen(status_path_template);
+  const int status_path_template_len = strlen(status_path_template) + 1;
   const int pid_len = strlen(pid);
   const int status_path_len = status_path_template_len + pid_len - 2;
 
   char status_path[status_path_len];
   char error_msg[BUF_SIZE];
 
-  Process_t *process = malloc(sizeof(Process_t));
-  process->pid = atoi(pid);
+  Process_t *p = (Process_t *)malloc(sizeof(Process_t));
```

Comment:
"p" as a variable name is too generic. try to use something like pProcess or similar. Hard to read and get the context.